WebNetSample/Controllers/ProductsController.cs

```diff
@@ -18,4 +23,47 @@ public async Task<IActionResult> Index()
 
         return View(productsWithDetails);
     }
+
+    public async Task<IActionResult> AllProducts()
+    {
+        var productsWithDetails = await _productService.GetListAsync(_paginationParameters);
+
+        return View(productsWithDetails);
+    }
+
+    [HttpGet]
+    public IActionResult Add()
+    {
+        Guid id = Guid.NewGuid();
+        DateTime date = DateTime.Now;
+
+        ViewBag.Id = id;
+        ViewBag.CreationDate = date;
+
+        return View();
+    }
+
+    [HttpPost]
+    public async Task<ActionResult> Add(Product product)
+    {
+        await _productService.AddAsync(product);
+
+        return RedirectToAction("Index");
+    }
+
+    [HttpGet]
+    public IActionResult Edit(string productToEdit)
```

Comment:
My bad: this endpoint should definitely accept only `id` of product. The reason is `GET` method - whenever you will click on browser refresh page - then you may loose your data + it may lead to data inconsistency from real model. So, let's pass here only id of product and take entity from DB. It will not lead to significant performance trouble because in real app one minor SQL query to search for entity by ID is not too dangerous.